Sources/SwiftTestingTestableCombinePublishers/SwiftTestingPublisherExpectation.swift

```diff
@@ -0,0 +1,674 @@
+        .init(upstreamPublisher: self).expectNotFailure(failure, message: message, sourceLocation: sourceLocation)
+    }
+    
+    /// Invokes the provided assertion closure on the `Failure` result's associated `Error` value  of the `Publisher`
+    /// Useful for calling `#expect` variants where custom evaluation is required
+    /// - Parameters:
+    ///   - assertion: The assertion to be performed on the `Failure` result's associated `Error` value
+    ///   - sourceLocation: The calling source. Used for showing context-appropriate unit test failures in Xcode
+    /// - Returns: A chainable `SwiftTestingPublisherExpectation` that matches the contextual upstream `Publisher` type
+    func expectFailure(_ assertion: @escaping (Failure) -> Void, sourceLocation: SourceLocation = #_sourceLocation) -> SwiftTestingPublisherExpectation<Self> {
+        .init(upstreamPublisher: self).expectFailure(assertion, sourceLocation: sourceLocation)
+    }
+}
+
+// MARK: - Void Publisher Expectations
+
+public extension SwiftTestingPublisherExpectation where UpstreamPublisher.Output == Void {
+    
+    /// Asserts that `Void` will be emitted by the `Publisher` one or more times
+    /// - Parameters:
+    ///   - sourceLocation: The calling source. Used for showing context-appropriate unit test failures in Xcode
+    /// - Returns: A chainable `PublisherExpectation` that matches the contextual upstream `Publisher` type
+    func expectVoid(sourceLocation: SourceLocation = #_sourceLocation) -> Self {
+        let expectation = SwiftTestingExpectation(description: "expectVoid()",
+                                                  sourceLocation: sourceLocation)
+        expectations.append(expectation)
+        upstreamPublisher
+            .sink { completion in
+            } receiveValue: { value in
+                expectation.fulfill()
+            }
+            .store(in: &cancellables)
+        return self
+    }
+}
+
+// MARK: - Publisher Extension for Void Publisher Expectations
+public extension Publisher where Output == Void {
+    
+    /// Asserts that `Void` will be emitted by the `Publisher` one or more times
+    /// - Parameters:
+    ///   - sourceLocation: The calling source. Used for showing context-appropriate unit test failures in Xcode
+    /// - Returns: A chainable `SwiftTestingPublisherExpectation` that matches the contextual upstream `Publisher` type
+    func expectVoid(sourceLocation: SourceLocation = #_sourceLocation) -> SwiftTestingPublisherExpectation<Self> {
+        .init(upstreamPublisher: self).expectVoid(sourceLocation: sourceLocation)
+    }
+}
+
+// MARK: - Private
+
+private extension SwiftTestingPublisherExpectation {
+    func waitForExpectationToBeFulfilled(_ expectation: SwiftTestingExpectation) async throws -> SwiftTestingExpectation {
+        while !expectation.isFulfilled {
+            // Wait for 0.1 seconds
+            try await Task.sleep(nanoseconds: 100_000_000)
```

Comment:
Yes, `Task.sleep` already does that, as documented [here](https://github.com/swiftlang/swift-evolution/blob/main/proposals/0304-structured-concurrency.md#voluntary-suspension).

>The sleep function accepts a plain integer as nanoseconds to sleep for, which mirrors known top-level functions performing the same action in the synchronous world. It will throw CancellationError if the task is cancelled while it sleeps, without waiting for the full sleep duration.